animation_nodes/nodes/gpencil/gp_layer_info.py

```diff
@@ -0,0 +1,136 @@
+        elif self.frameType == "FRAME":
+            if self.useFrameList:
+                return "execute_FrameNumbers"
+            else:
+                return "execute_FrameNumber"
+        elif self.frameType == "ALL":
+            return "execute_AllFrames"
+
+    def execute_ActiveFrame(self, layer, scene):
+        frames = layer.frames
+        if len(layer.frames) == 0:
+            return GPFrame(), layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+        return (self.getActiveFrame(scene.frame_current, frames), layer.layerName, layer.blendMode,
+                layer.opacity, layer.passIndex)
+
+    def execute_FrameIndex(self, layer, frameIndex):
+        frames = layer.frames
+        if len(layer.frames) == 0:
+            return GPFrame(), layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+        frame = self.getFrame(frames, frameIndex)
+        return frame, layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+    def execute_FrameIndices(self, layer, frameIndices):
+        frames = layer.frames
+        if len(layer.frames) == 0:
+            return [], layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+        outFrames = [self.getFrame(frames, index) for index in frameIndices]
+        return outFrames, layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+    def execute_FrameNumber(self, layer, frameNumber):
+        frames = layer.frames
+        if len(layer.frames) == 0:
+            return GPFrame(), layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+        frameNumbers = self.getFrameNumbers(frames)
+        index = self.getFrameNumberIndex(frameNumbers, frameNumber)
+        return frames[index], layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+    def execute_FrameNumbers(self, layer, inFrameNumbers):
+        frames = layer.frames
+        if len(frames) == 0:
+            return [], layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+        frameNumbers = self.getFrameNumbers(frames)
+        outFrames = [frames[self.getFrameNumberIndex(frameNumbers, frameNumber)] for frameNumber in inFrameNumbers]
+        return outFrames, layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+    def execute_AllFrames(self, layer):
+        frames = layer.frames
+        return frames, layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+
+    def getActiveFrame(self, currentFrame, frames):
+        frameNumbers = self.getFrameNumbers(frames)
+        if currentFrame <= frameNumbers.getMinValue(): return frames[0]
+        allFrames = list(filter(lambda x: x <= currentFrame, frameNumbers))
+        return frames[frameNumbers.index(allFrames[-1])]
+
+    def getFrameNumbers(self, frames):
+        frameNumbers = LongList(length = len(frames))
+        for i, frame in enumerate(frames):
+            frameNumbers[i] = frame.frameNumber
+        return frameNumbers
```

Comment:
This function can be inlined. And a list comprehension can be used.

Reply:
This function is no longer needed.